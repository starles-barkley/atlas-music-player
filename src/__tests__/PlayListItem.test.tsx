import { render } from "@testing-library/react";
import { expect, test, vi } from "vitest";
import PlaylistItem from "../components/PlaylistItem.tsx";
import { PlayerProvider } from "../components/PlayerContext.tsx";


test("playlist item returns title, artist, and time", () => {
    const { container } = render(
    <PlayerProvider>
        <PlaylistItem
        songtitle="Painted in Blue"
        artist="Soul Canvas"
        time="194" onClick={() => {}}
            />
        </PlayerProvider>
    );
    expect(container).toMatchSnapshot();
});

test("playlist item onClick prop function was not called", () => {
    const mockCurrentSongId = vi.fn();

    const { container } = render(
        <PlayerProvider>
            <PlaylistItem
            songtitle="Painted in Blue"
            artist="Soul Canvas"
            time="194"
            onClick={mockCurrentSongId}
            />
        </PlayerProvider>
    );
    expect(container).toMatchSnapshot;
    expect(mockCurrentSongId).not.toHaveBeenCalled();
})