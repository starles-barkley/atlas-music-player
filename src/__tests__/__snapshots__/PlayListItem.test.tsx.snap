// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Playlist Component > matches snapshot 1`] = `
<div>
  <div
    class="hover:bg-transparent hover:outline hover:outline-2 hover:outline-offset-8 hover:outline-primary hover:text-primary p-1 flex w-full cursor-pointer flex-col rounded-lg "
  >
    <p
      class="ml-2 mr-2 font-comfortaa font-bold text-text"
    />
    <div
      class="flex w-full justify-between"
    >
      <p
        class="ml-2 mr-2 font-montserrat text-text"
      />
      <p
        class="ml-2 mr-2 font-montserrat text-textMuted"
      >
        NaN:NaN
      </p>
    </div>
  </div>
</div>
`;